### PR TITLE
Create Federations struct deserialization from toml

### PR DESCRIPTION
**Pull request by rantan**
This PR includes #93 changes.

### CODE REVIEW

**Comment by rantan** [comment]
rebased on #93

**Comment by Yamaguchi** [comment]
Can you rebase from origin/master?

**Comment by rantan** [comment]
Rebased on master branch.

File: src/federation.rs [hunk 1]
```diff
@@ -192,6 +210,30 @@ impl Federation {
...
+        SerFederation {
+            block_height: self.block_height,
+            threshold: self.threshold,
+            nodevss: self.nodevss,
+        }
+    }
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct SerFederations {
```
Comment (Yamaguchi) [inline src/federation.rs:228]: I think SerFederations or SerFederation is not needed if we use Federations or Federation instead and change the federation file format.

Comment (rantan) [reply src/federation.rs:228]: I think it is appropriate idea for current implementation but I'm wondering that the Federation struct would have some new fields for caching calculated value like an aggregated public key.

We have some options for that. For example add the cache fields to Federation and set attribute for skipping serialization on serde functionality. 
However I thought the way this PR is using was most simple for me.

**Pull request merged**